tests/components/heos/test_media_player.py

```diff
@@ -162,67 +162,142 @@
+        assert player.play_previous.call_count == 1
+        player.play_previous.reset_mock()
+        player.play_previous.side_effect = CommandError(None, "Failure", 1)
+    assert "Unable to move to previous track: Failure (1)" in caplog.text
+
+
+async def test_next_track(hass, config_entry, config, controller, caplog):
+    """Test the next track service."""
+    await setup_platform(hass, config_entry, config)
+    player = controller.players[1]
+    # First pass completes successfully, second pass raises command error
+    for _ in range(2):
+        await hass.services.async_call(
+            MEDIA_PLAYER_DOMAIN, SERVICE_MEDIA_NEXT_TRACK,
+            {ATTR_ENTITY_ID: 'media_player.test_player'}, blocking=True)
+        assert player.play_next.call_count == 1
+        player.play_next.reset_mock()
+        player.play_next.side_effect = CommandError(None, "Failure", 1)
+    assert "Unable to move to next track: Failure (1)" in caplog.text
+
+
+async def test_stop(hass, config_entry, config, controller, caplog):
+    """Test the stop service."""
+    await setup_platform(hass, config_entry, config)
+    player = controller.players[1]
+    # First pass completes successfully, second pass raises command error
+    for _ in range(2):
+        await hass.services.async_call(
+            MEDIA_PLAYER_DOMAIN, SERVICE_MEDIA_STOP,
+            {ATTR_ENTITY_ID: 'media_player.test_player'}, blocking=True)
+        assert player.stop.call_count == 1
+        player.stop.reset_mock()
+        player.stop.side_effect = CommandError(None, "Failure", 1)
+    assert "Unable to stop: Failure (1)" in caplog.text
+
+
+async def test_volume_mute(hass, config_entry, config, controller, caplog):
+    """Test the volume mute service."""
+    await setup_platform(hass, config_entry, config)
+    player = controller.players[1]
+    # First pass completes successfully, second pass raises command error
+    for _ in range(2):
+        await hass.services.async_call(
+            MEDIA_PLAYER_DOMAIN, SERVICE_VOLUME_MUTE,
+            {ATTR_ENTITY_ID: 'media_player.test_player',
+             ATTR_MEDIA_VOLUME_MUTED: True}, blocking=True)
+        assert player.set_mute.call_count == 1
+        player.set_mute.reset_mock()
+        player.set_mute.side_effect = CommandError(None, "Failure", 1)
+    assert "Unable to set mute: Failure (1)" in caplog.text
+
+
+async def test_shuffle_set(hass, config_entry, config, controller, caplog):
+    """Test the shuttle set service."""
+    await setup_platform(hass, config_entry, config)
+    player = controller.players[1]
+    # First pass completes successfully, second pass raises command error
+    for _ in range(2):
+        await hass.services.async_call(
+            MEDIA_PLAYER_DOMAIN, SERVICE_SHUFFLE_SET,
+            {ATTR_ENTITY_ID: 'media_player.test_player',
+             ATTR_MEDIA_SHUFFLE: True}, blocking=True)
+        player.set_play_mode.assert_called_once_with(player.repeat, True)
```

Comment:
We can avoid the risk of typo passing a bad test by using `assert mocked.call_count == 1` and `assert mocked.call_args == mock.call(...)`

Reply:
Isn't that what `assert_called_once_with` is equivalent to?  Not sure I'm following...

Reply:
If we typo that method, it will pass the test whatever we put in as arguments, right?

Reply:
`player.repeat` is a property.  The only part we care about is the `True` as that matches the `ATTR_MEDIA_SHUFFLE` in the service call.  HEOS has a single command to set both repeat and shuffle -- since media player only supports shuffle, we pass the current value for `repeat` to ensure it doesn't get changed when we set shuffle.

Reply:
I'm talking about `mocked.assert_called_once_with`. If we make a typo there, it will be treated as an attribute/method of the mock.

Reply:
The mock is created using `player = Mock(HeosPlayer, autospec=True)`.  [According to the docs](https://docs.python.org/3.5/library/unittest.mock.html#unittest.mock.Mock), when the `spec` parameter is used (i.e. `HeosPlayer`), `AttributeError` is raised when accessing an attribute not in the class.  I'll confirm when I get home...

FYI the `autospec` is extraneous -- evidently that's only for `patch`.  I'll remove that...

Reply:
Confirmed it does raise `AttributeError` and removed `autospec` in latest commit.